ambry-api/src/main/java/com/github/ambry/quota/QuotaEnforcer.java

```diff
@@ -36,7 +36,16 @@
    * @param requestCostMap {@link Map} of {@link QuotaName} to the cost incurred to handle the request.
    * @return QuotaRecommendation object with the recommendation.
    */
-  QuotaRecommendation chargeAndRecommend(RestRequest restRequest, BlobInfo blobInfo, Map<QuotaName, Double> requestCostMap);
+  QuotaRecommendation chargeAndRecommend(RestRequest restRequest, BlobInfo blobInfo,
+      Map<QuotaName, Double> requestCostMap);
+
+  /**
+   * Makes an {@link QuotaRecommendation} using the information in {@link RestRequest}. This
+   * method also charges one unit of request cost against the quota.
```

Comment:
How does this method know which type of quota to charge against (usually given by QuotaName)?

Reply:
It looks at RestRequest's `getRestMethod` and converts it to write or read operation. For writes, we can charge for storage quota and WCU. For reads, we only charge for RCU.